testing/tests/DevExpress.ui.widgets/treeViewParts/expandedItems.js

```diff
@@ -690,4 +692,24 @@ module("Expanded items", {
 
         assert.equal(contentReadyStub.callCount, 2, "event is thrown twice");
     });
+
+    ['dataSource', 'items'].forEach((optionName) => {
+        QUnit.test(`Nodes expanding works even with loop/cycle in ${optionName} option`, function(assert) {
+            let options = { dataStructure: "plain", rootValue: 1 };
+            options[optionName] = [
+                { id: 1, text: "item1", parentId: 2, selected: false, expanded: false },
```

Comment:
How 'expandItem' will work with 'expanded: true' config?

Reply:
It tries to set active node state.
If it already has active state, it does nothing.

![image](https://user-images.githubusercontent.com/57402891/69343580-f2f86880-0c7e-11ea-9aea-1dc0d587208e.png)

I have also add more test cases